x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,222 @@
+---
+# This template serves as a foundation for indices and data streams using ECS.
+template:
+
+  settings:
+    # This is a made up setting which does not exist today. The idea is that
+    # any field that shows up after the first 1000 fields is kept in source
+    # but does not show up as runtime for index field
+    #index.mapping.total_fields.soft_limit: 1000
+
+    # Ignores any malformed data. Data will still be persisted in _source
+    index.mapping.ignore_malformed": true
+
+  mappings:
+
+    # Any field that is discovered is mapped as a runtime field
+    dynamic: runtime
+
+    # DISCUSS: What are the pros / cons of having this enabled?
+    date_detection: false
+
+    ### All the properties which are set in each template
+    properties:
+
+      # This ensures all subobjects are flattened and prevents and object / keyword conflicts
+      # As subobject is set to false, no objects like host, error or similar are mapped.
+      subobject: false
+
+      # @timestamp is the only field that MUST exist.
+      "@timestamp":
+        type: date
+        ignore_malformed: false
+
+      # The data stream properties are required in any index following the data stream naming scheme
+      # But not all data streams or indices which use ECS are part of the data stream naming scheme.
+      # As an example logstash-* and *beats-*. Because of this, the data_stream.* fields are not part
+      # of this template. It is expected that these are set by another template for the data stream
+      # naming scheme.
+      #data_stream.type:
+      #  type: constant_keyword
+      #  #value: logs
+      #data_stream.dataset:
+      #  type: constant_keyword
+      #data_stream.namespace:
+      #  type: constant_keyword
+
+
+    # To ensure no empty mappings are in an index, everything except the properties above are mapped as dynamic templates
+    # The dynamic mappings here are a best effort to map ECS core fields: https://github.com/elastic/ecs/blob/8.4/generated/csv/fields.csv
+    # As most fields are keyword, these will be automatically be matched correctly. Only ECS core fields were taken into account.
+    dynamic_templates:
+
+      ### Index mappings
+
+      # Message field is always index
+      # DISCUSS: Should only the top level message field always be indexed?
+      - match_message:
+          path_match: message
+          mapping:
+            type: match_only_text
+
+      ### Runtime mappings
+
+      # Set the ECS version
+      - match_ecs.version:
+          path_match: ecs.version
+          # DISCUSS: In ECS this is still a keyword field
+          type: version
```

Comment:
This is mapped as an indexed field but is under the `### Runtime mappings` section. The version type is currently not supported on runtime fields.

Reply:
I don't think the ECS version should be an indexed field as I have not seen lots of query on it. I'll add it as runtime field.

Even in ECS today it is a keyword as by the time it was introduced, version did not exist in Elasticsearch. The simplest is to just map it to keyword as this is being supported by Elasticsearch and is what is in ECS. But is it future proof?